libs/ui-lib/lib/cim/components/Hypershift/HostedClusterWizard/HostsStep/NodePoolForm.tsx

```diff
@@ -63,20 +69,44 @@ const NodePoolForm: React.FC<NodePoolFormProps> = ({ infraEnvs, agents, index, o
   let previousNodePoolsCount = 0;
 
   for (let i = 0; i < index; i++) {
-    previousNodePoolsCount += values.nodePools[i].count;
+    previousNodePoolsCount += (
+      values.nodePools[i].useAutoscaling
+        ? values.nodePools[i].autoscaling?.maxReplicas
+        : values.nodePools[i].count
+    ) as number;
   }
 
   const maxAgents = Math.max(
     0,
     Math.min(matchingAgents.length, availableAgents.length - previousNodePoolsCount),
   );
 
-  const currentCount = values.nodePools[index].count;
+  const currentCount = (
+    values.nodePools[index].useAutoscaling
+      ? values.nodePools[index].autoscaling?.maxReplicas
+      : values.nodePools[index].count
+  ) as number;
+
   React.useEffect(() => {
```

Comment:
this `useEffect` has a few issues
 - it does not check `useAutoscaling` to determine which form value to update
 - the whole autoscaling branch is flawed (ie setting min value only if max agents === 0, but what about max being less than min)
 - generally I think this useEffect which is changing values without notifying user was not a good decision

We should fix the first two. The third one, which is a bigger change, can be deferred.

Reply:
I fixed the first two. I also added validation on the autoscaling fields that triggers an error when the numbers are <1.

![image](https://github.com/openshift-assisted/assisted-installer-ui/assets/87187179/301115d8-ff96-4a50-a202-290d8321a6fd)